libraries/chain/include/scorum/chain/services/comment.hpp

```diff
@@ -23,6 +23,8 @@ struct comment_service_i : public base_service_i<comment_object>
 
     virtual comment_refs_type get_children(const account_name_type& parent_author,
                                            const std::string& parent_permlink) const = 0;
+
+    virtual void set_rewarded_flag(const comment_object& comment) = 0;
```

Comment:
If you remove `rewarded` member than there will be no need of such service method.